cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -229,21 +229,36 @@ func (c *CStorPoolController) cStorPoolAddEvent(cStorPoolGot *apis.CStorPool) (s
 	}
 	var importPoolErr error
 	var status string
+	var importOptions pool.ImportOptions
 	cachefileFlags := []bool{true, false}
 	for _, cachefileFlag := range cachefileFlags {
-		status, importPoolErr = c.importPool(cStorPoolGot, cachefileFlag)
+		importOptions.CachefileFlag = cachefileFlag
+		status, importPoolErr = c.triggerImportPool(cStorPoolGot, &importOptions)
 		if status == string(apis.CStorPoolStatusOnline) {
-			c.recorder.Event(cStorPoolGot, corev1.EventTypeNormal, string(common.SuccessImported), string(common.MessageResourceImported))
-			common.SyncResources.IsImported = true
 			return status, nil
 		}
 		if importPoolErr != nil {
-			glog.Errorf("failed to handle add event: import succeded with failed operations %v", importPoolErr)
-			c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.FailureImported), string(common.FailureImportOperations))
 			return status, importPoolErr
 		}
 	}
 
+	devIDList, err := c.getDeviceIDs(cStorPoolGot)
+	if err != nil {
+		return string(apis.CStorPoolStatusOffline), errors.Wrapf(err, "failed to get device id of disks for csp %s", cStorPoolGot.Name)
+	}
+
+	devPath := pool.GetDevPath(devIDList[0])
```

Comment:
Why is only the first devID  taken into consideration and not other ?

Reply:
this is also under assumption that all devices of the pool are in same directory..